internal/integrations/v4/integration/integration_test.go

```diff
@@ -163,6 +164,33 @@ func TestTimeout_Default(t *testing.T) {
 	// THEN the integration has a default timeout
 	assert.Equal(t, defaultTimeout, i.Timeout)
 }
+func TestInterval_EnvironmentVariableCustom(t *testing.T) {
+	// GIVEN a configuration with custom interval
+	// WHEN an integration is loaded from it
+	i, err := NewDefinition(config2.ConfigEntry{InstanceName: "foo", Interval: "55s", Exec: config2.ShlexOpt{"bar"}}, ErrLookup, nil, nil)
+	require.NoError(t, err)
+
+	// THEN the integration has an environment variable with the interval value
+	assert.Equal(t, "55s", i.ExecutorConfig.Environment[intervalEnvVarName])
+}
+func TestInterval_EnvironmentVariableCero(t *testing.T) {
+	// GIVEN a configuration with custom interval
+	// WHEN an integration is loaded from it
+	i, err := NewDefinition(config2.ConfigEntry{InstanceName: "foo", Interval: "0", Exec: config2.ShlexOpt{"bar"}}, ErrLookup, nil, nil)
```

Comment:
could you use this constructor https://github.com/newrelic/infrastructure-agent/blob/30d0e84f3526ab0e1398d7908cc6e73193a21109/internal/agent/cmdchannel/runintegration/runintegration.go#L91 so we ensure it's the same output command-channel generates?

Reply:
which is https://github.com/newrelic/infrastructure-agent/blob/30d0e84f3526ab0e1398d7908cc6e73193a21109/internal/agent/cmdchannel/runintegration/runintegration.go#L63

Reply:
@varas  there is an import cycle error to add this here. Would be ok if I add a check to [this test](https://github.com/newrelic/infrastructure-agent/blob/master/internal/agent/cmdchannel/runintegration/runintegration_test.go#L54) to verify the interval is 0?

Reply:
`assert.Equal(t, time.Duration(0), d.Interval)`

Reply:
I did this

Reply:
sure, fine with me